src/screens/ConnectScreen.js

```diff
@@ -321,11 +335,20 @@ export default class ConnectScreen extends React.Component {
    * rocket function to bind the onPress method
    */
   renderVeteranMarkers() {
-    return this.state.veterans.map((veteran) => {
+    const locations = new Set();
+    let currentVets = this.state.currentVets;
+    if (this.state.currentVets.length + this.state.currentPos.length > 10) {
+      currentVets = this.state.currentVets.slice(0, 10 - this.state.currentPos.length)
```

Comment:
This is not the best

Reply:
we should probably have a global constant variable so you don't have 10s everywhere in the code